pkg/server/share_manager.go

```diff
@@ -210,6 +266,107 @@ func (m *ShareManager) resizeVolume(devicePath, mountPath string) error {
 	return nil
 }
 
+func (m *ShareManager) getEnvAsInt(key string, defaultVal int) int {
+	env := os.Getenv(key)
+	if env == "" {
+		m.logger.Warnf("Failed to get expected environment variable, env %v wasn't set, defaulting to %v", key, defaultVal)
+		return defaultVal
+	}
+	value, err := strconv.Atoi(env)
+	if err != nil {
+		m.logger.Warnf("Failed to convert environment variable, %v, value %v, to an int, defaulting to %v", key, env, defaultVal)
+		return defaultVal
+	}
+	return value
+}
+
+func (m *ShareManager) getEnvAsBool(key string, defaultVal bool) bool {
+	env := os.Getenv(key)
+	if env == "" {
+		m.logger.Warnf("Failed to get expected environment variable, env %v wasn't set, defaulting to %v", key, defaultVal)
+		return defaultVal
+	}
+	value, err := strconv.ParseBool(env)
+	if err != nil {
+		m.logger.Warnf("Failed to convert environment variable, %v, value %v, to an int, defaulting to %v", key, env, defaultVal)
+		return defaultVal
+	}
+	return value
+}
+
+func (m *ShareManager) NewKubeClient() (*kubernetes.Clientset, error) {
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	clientset, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return clientset, nil
+}
+
+func (m *ShareManager) takeLease() error {
+	if m.leaseClient == nil {
+		return fmt.Errorf("kubernetes API client is unset")
```

Comment:
```suggestion
		return fmt.Error("kubernetes API client is unset")
```

Reply:
Hmm.  There is no `fmt.Error`, and `go vet` and our linter are both happy with an empty variadic list.
It could be changed to `errors.New` and give a stack trace for no extra money.  I think I'll leave it as is, unless you strongly prefer otherwise.

Reply:
It is logrus that has <word> and <word>f variants. Got confused.